workers/python/kitchen_sink.py

```diff
@@ -1,269 +1,173 @@
-    value: Any
-    run_id: bool
-
-
-@dataclass
-class ErrorAction:
-    message: str
-    details: Any
-    attempt: bool
-
-
-@dataclass
-class ContinueAsNewAction:
-    while_above_zero: int
-
-
-@dataclass
-class SleepAction:
-    millis: int
-
-
-@dataclass
-class QueryHandlerAction:
-    name: str
-
-
-@dataclass
-class SignalAction:
-    name: str
-
-
-@dataclass
-class ExecuteActivityAction:
-    name: str
-    task_queue: str
-    args: Optional[Sequence[Any]]
-    count: int
-    index_as_arg: bool
-    schedule_to_close_timeout_ms: int
-    start_to_close_timeout_ms: int
-    schedule_to_start_timeout_ms: int
-    cancel_after_ms: int
-    wait_for_cancellation: bool
-    heartbeat_timeout_ms: int
-    retry_max_attempts: int
-    non_retryable_error_types: Optional[Sequence[str]]
-
-
-@dataclass
-class ExecuteChildWorkflowAction:
-    name: Optional[str]
-    count: int
-    args: Optional[Sequence[Any]]
-    params: Optional[WorkflowParams]
-
-
-def workflow_query(arg: str) -> str:
-    return arg
+from protos.kitchen_sink_pb2 import (
+    Action,
+    ActionSet,
+    ActivityCancellationType,
+    ExecuteActivityAction,
+    WorkflowInput,
+)
 
 
 @workflow.defn(name="kitchenSink")
 class KitchenSinkWorkflow:
-    @workflow.run
-    async def run(self, input: Optional[WorkflowParams]) -> Any:
-        workflow.logger.info("Started kitchen sink workflow {}".format(input))
-        if not input:
-            return None
-        if input.action_set and input.action_set.actions:
-            should_return, return_value = await self.handle_action_set(
-                input, input.action_set
-            )
-            if should_return:
-                return return_value
+    action_set_queue: asyncio.Queue[ActionSet] = asyncio.Queue()
 
-        if input.action_set_signal:
-            action_set_queue: asyncio.Queue[ActionSet] = asyncio.Queue()
+    @workflow.signal
+    async def do_actions_signal(self, action_set: ActionSet) -> None:
+        self.action_set_queue.put_nowait(action_set)
 
-            def enqueue(item: ActionSet) -> None:
-                action_set_queue.put_nowait(item)
-
-            workflow.set_signal_handler(input.action_set_signal, enqueue)
-            while True:
-                action_set = await action_set_queue.get()
-                should_return, return_value = await self.handle_action_set(
-                    input, action_set
-                )
-                if should_return:
+    @workflow.run
+    async def run(self, input: Optional[WorkflowInput]) -> Any:
+        workflow.logger.info("Started kitchen sink workflow")
+
+        # Run all initial input actions
+        if input and input.initial_actions:
+            for action_set in input.initial_actions:
+                return_value = await self.handle_action_set(action_set)
+                if return_value is not None:
                     return return_value
 
-        return return_value
+        # Run all actions from signals
```

Comment:
Not directly related to your PR, but I think this approach of handling signals is not as valuable as handling the actions in the handler itself. There have been quite a few bugs in Java where something is broken in a signal handler, but not in the main workflow thread.